### DISCUSSION TITLE
Resolved: Can't delete a folder (the Music app interfered)

### DISCUSSION THREAD

**Discussion by Bruceumba** [nextcloud/all-in-one#2999] [Questions] [answered]
Resolved. It was the Music app that was creating the error.

Here is the error I get when attempting to delete a folder. I can change its name. I get the feeling that one of my apps is interacting oddly with the directory manager, as has happened in the past. The folder is named "Zdelete". v6.2.1

[webdav] 
Error: TypeError: OC\Files\Cache\QuerySearchHelper::getCachesAndMountPointsForSearch(): Argument #1 ($root) must be of type OC\Files\Node\Root, OC\Files\Node\LazyRoot given, called in /var/www/html/lib/private/Files/Node/Folder.php on line 237 at <<closure>>

 0. /var/www/html/lib/private/Files/Node/Folder.php line 237
    OC\Files\Cache\QuerySearchHelper->getCachesAndMountPointsForSearch(["OC\\Files\\Node\\LazyRoot"], "/bruce/files/Zdelete", false)
 1. /var/www/html/lib/private/Files/Node/Folder.php line 291
    OC\Files\Node\Folder->search(["OC\\Files\\Search\\SearchQuery"])
 2. /var/www/html/custom_apps/music/lib/Utility/Scanner.php line 383
    OC\Files\Node\Folder->searchByMime("audio")
 3. /var/www/html/custom_apps/music/lib/Hooks/FileHooks.php line 42
    OCA\Music\Utility\Scanner->deleteFolder(["OC\\Files\\Node\\Folder"])
 4. <<closure>>
    OCA\Music\Hooks\FileHooks::deleted(["OC\\Files\\Node\\Folder"])
 5. /var/www/html/lib/private/Hooks/EmitterTrait.php line 105
    call_user_func_array(["OCA\\Music\\Ho ... "], [["OC\\Files\\Node\\Folder"]])
 6. /var/www/html/lib/private/Hooks/PublicEmitter.php line 40
    OC\Hooks\BasicEmitter->emit("\\OC\\Files", "preDelete", [["OC\\Files\\Node\\Folder"]])
 7. /var/www/html/lib/private/Files/Node/Root.php line 143
    OC\Hooks\PublicEmitter->emit("\\OC\\Files", "preDelete", [["OC\\Files\\Node\\Folder"]])
 8. <<closure>>
    OC\Files\Node\Root->emit("\\OC\\Files", "preDelete", [["OC\\Files\\Node\\Folder"]])
 9. /var/www/html/lib/private/Files/Node/LazyFolder.php line 72
    call_user_func_array([["OC\\Files\\Node\\Root"],"emit"], ["\\OC\\Files"," ... ]])
10. /var/www/html/lib/private/Files/Node/LazyFolder.php line 100
    OC\Files\Node\LazyFolder->__call("emit", ["\\OC\\Files"," ... ]])
11. /var/www/html/lib/private/Files/Node/HookConnector.php line 145
    OC\Files\Node\LazyFolder->emit("\\OC\\Files", "preDelete", [["OC\\Files\\Node\\Folder"]])
12. /var/www/html/lib/private/legacy/OC_Hook.php line 105
    OC\Files\Node\HookConnector->delete([true,"/Zdelete"])
13. /var/www/html/lib/private/Files/View.php line 1295
    OC_Hook::emit("OC_Filesystem", "delete", [true,"/Zdelete"])
14. /var/www/html/lib/private/Files/View.php line 1165
    OC\Files\View->runHooks(["delete"], "/Zdelete")
15. /var/www/html/lib/private/Files/View.php line 351
    OC\Files\View->basicOperation("rmdir", "/Zdelete", ["delete"])
16. /var/www/html/apps/dav/lib/Connector/Sabre/Directory.php line 307
    OC\Files\View->rmdir("/Zdelete")
17. /var/www/html/3rdparty/sabre/dav/lib/DAV/Tree.php line 179
    OCA\DAV\Connector\Sabre\Directory->delete()
18. /var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 281
    Sabre\DAV\Tree->delete("files/bruce/Zdelete")
19. /var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpDelete(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
20. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit("method:DELETE", [["Sabre\\HTTP\\ ... ]])
21. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
22. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
23. /var/www/html/apps/dav/lib/Server.php line 366
    Sabre\DAV\Server->exec()
24. /var/www/html/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
25. /var/www/html/remote.php line 172
    require_once("/var/www/html/a ... p")

DELETE /remote.php/dav/files/bruce/Zdelete
from 192.168.0.1 by bruce at 2023-07-21T04:28:27+00:00

Any ideas what app I might disable to delete this folder?

**Comment by szaimen** [comment, marked as answer]
> Resolved. It was the Music app that was creating the error.
